File: src/main/java/rd/sb_ng_browsergames/model/MinesweeperScore.java
package rd.sb_ng_browsergames.model;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name="minesweeper_scores")
public class MinesweeperScore {

    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    @Column(name="id")
    private long id;

    @Column(name="player")
    private String userName;

    @Column(name="score")
    private int score;

    @Column(name="table_size")
    private String tableSize;

    @Column(name="date")
    private LocalDateTime date;

    
    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public int getScore() {
        return score;
    }

    public void setScore(int score) {
        this.score = score;
    }

    public String getTableSize() {
        return tableSize;
    }

    public void setTableSize(String tableSize) {
        this.tableSize = tableSize;
    }

    public LocalDateTime getDate() {
        return date;
    }

    public void setDate(LocalDateTime date) {
        this.date = date;
    }

    
}

File: src/app/model/minesweeper-score.ts
export class MinesweeperScore {

    private _id: number;
    private _userName: string;
    private _score: number;
    private _tableSize: string;
    private _date: Date;


    constructor(id: number, userName: string, score: number, tableSize: string, date: Date) {
        this._id = id;
        this._userName = userName;
        this._score = score;
        this._tableSize = tableSize;
        this._date = date;
    }


    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }

    public get userName(): string {
        return this._userName;
    }
    public set userName(value: string) {
        this._userName = value;
    }

    public get score(): number {
        return this._score;
    }
    public set score(value: number) {
        this._score = value;
    }

    public get tableSize(): string {
        return this._tableSize;
    }
    public set tableSize(value: string) {
        this._tableSize = value;
    }

    public get date(): Date {
        return this._date;
    }
    public set date(value: Date) {
        this._date = value;
    }


    // public getId(): number {
    //     return this.id;
    // }

    // public getUserName(): string {
    //     return this.userName;
    // }

    // public getScore(): number {
    //     return this.score;
    // }

    // public getTableSize(): string {
    //     return this.tableSize;
    // }

    // public getDate(): Date {
    //     return this.date;
    // }

    // public setUserName(userName: string) {
    //     this.userName = userName;
    // }

}
